### ISSUE TITLE
[🐛 Bug]: Error setting name in helm release

### ISSUE THREAD

**Issue by maksim-paskal** [SeleniumHQ/docker-selenium#2006]
### What happened?

Error while installing helm release

```shell
helm template my-release-name ./charts/selenium-grid \
--set-string chromeNode.nameOverride=my-chrome-name \
--set-string hub.nameOverride=my-hub-name \
| kubectl apply --dry-run=server -f -
```

### Relevant log output

```shell
The Deployment "my-chrome-name" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{"app":"my-chrome-name", "app.kubernetes.io/component":"selenium-grid-4.15.0-20231102", "app.kubernetes.io/instance":"my-release-name", "app.kubernetes.io/managed-by":"helm", "app.kubernetes.io/name":"my-chrome-name", "app.kubernetes.io/version":"4.15.0-20231102", "helm.sh/chart":"selenium-grid-0.24.0"}: `selector` does not match template `labels`
```

### Helm version
```
version.BuildInfo{Version:"v3.11.3", GitCommit:"323249351482b3bbfc9f5004f65d400aa70f9ae7", GitTreeState:"clean", GoVersion:"go1.20.3"}
```

**Comment by maksim-paskal** [comment]
@vietnd96 this issue goes after your [commit](https://github.com/SeleniumHQ/docker-selenium/pull/1989/commits/8427e43c5082a5565dd7c63e32485abe22f22404)

How about to use name with it static name?

**Comment by VietND96** [comment]
Hi @maksim-paskal , could you please confirm that when `.nameOverride` is set, all resources name belong to particular component will use that name accordingly? For example when `helm upgrade -i my-release-name --set-string chromeNode.nameOverride=my-chrome-name --set-string
 hub.nameOverride=my-hub-name .\selenium-grid-0.24.1.tgz`
kubectl get pods
```bash
NAME                                     READY   STATUS    RESTARTS   AGE   LABELS
my-chrome-name-84dd7bf974-sd29p          1/1     Running   0          34s   app.kubernetes.io/component=selenium-grid-4.15.0-20231110,app.kubernetes.io/instance=my-release-name,app.kubernetes.io/managed-by=helm,app.kubernetes.io/name=my-chrome-name,app.kubernetes.io/version=4.15.0-20231110,app=my-chrome-name,helm.sh/chart=selenium-grid-0.24.1,pod-template-hash=84dd7bf974
my-hub-name-5dc96c99d9-rph2v             1/1     Running   0          34s   app.kubernetes.io/component=selenium-grid-4.15.0-20231110,app.kubernetes.io/instance=my-release-name,app.kubernetes.io/managed-by=helm,app.kubernetes.io/name=my-hub-name,app.kubernetes.io/version=4.15.0-20231110,app=my-hub-name,helm.sh/chart=selenium-grid-0.24.1,pod-template-hash=5dc96c99d9
selenium-edge-node-5b6657478c-q5q6l      1/1     Running   0          34s   app.kubernetes.io/component=selenium-grid-4.15.0-20231110,app.kubernetes.io/instance=my-release-name,app.kubernetes.io/managed-by=helm,app.kubernetes.io/name=selenium-edge-node,app.kubernetes.io/version=4.15.0-20231110,app=selenium-edge-node,helm.sh/chart=selenium-grid-0.24.1,pod-template-hash=5b6657478c
selenium-firefox-node-74d4c5b465-h8frs   1/1     Running   0          34s   app.kubernetes.io/component=selenium-grid-4.15.0-20231110,app.kubernetes.io/instance=my-release-name,app.kubernetes.io/managed-by=helm,app.kubernetes.io/name=selenium-firefox-node,app.kubernetes.io/version=4.15.0-20231110,app=selenium-firefox-node,helm.sh/chart=selenium-grid-0.24.1,pod-template-hash=74d4c5b465
```
kubectl get svc
```bash
NAME                    TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                      AGE     LABELS
my-chrome-name          ClusterIP   10.111.80.35     <none>        6900/TCP                     5m31s   app.kubernetes.io/component=selenium-grid-4.15.0-20231110,app.kubernetes.io/instance=my-release-name,app.kubernetes.io/managed-by=Helm,app.kubernetes.io/version=4.15.0-20231110,helm.sh/chart=selenium-grid-0.24.1,name=my-chrome-name
my-hub-name             ClusterIP   10.100.217.109   <none>        4444/TCP,4442/TCP,4443/TCP   5m31s   app.kubernetes.io/component=selenium-grid-4.15.0-20231110,app.kubernetes.io/instance=my-release-name,app.kubernetes.io/managed-by=Helm,app.kubernetes.io/version=4.15.0-20231110,app=my-hub-name,helm.sh/chart=selenium-grid-0.24.1
selenium-edge-node      ClusterIP   10.111.74.203    <none>        6900/TCP                     5m31s   app.kubernetes.io/component=selenium-grid-4.15.0-20231110,app.kubernetes.io/instance=my-release-name,app.kubernetes.io/managed-by=Helm,app.kubernetes.io/version=4.15.0-20231110,helm.sh/chart=selenium-grid-0.24.1,name=selenium-edge-node
selenium-firefox-node   ClusterIP   10.99.180.116    <none>        6900/TCP                     5m31s   app.kubernetes.io/component=selenium-grid-4.15.0-20231110,app.kubernetes.io/instance=my-release-name,app.kubernetes.io/managed-by=Helm,app.kubernetes.io/version=4.15.0-20231110,helm.sh/chart=selenium-grid-0.24.1,name=selenium-firefox-node
```
If this outcome is correct your expectation, I will raise PR to fix this soon

**Comment by maksim-paskal** [comment]
@vietnd96 how can I get `selenium-grid-0.24.1.tgz` ? I already create [changes](https://github.com/SeleniumHQ/docker-selenium/compare/trunk...maksim-paskal:docker-selenium:fix-helm-chart-labels) to fix my issue, I want to know is it OK for you if I will use static name instead fullname func as you commited.

**Comment by VietND96** [comment]
@maksim-paskal , here is my change for `0.24.1` https://github.com/SeleniumHQ/docker-selenium/compare/trunk...ndviet:docker-selenium:trunk
In this change, I updated the template function in labels also. The difference is when `.overrideName` is set, it will affect wherever the static name was used before.
Can you please comment on which is better?
You can add this repo to get chart `0.24.1` contains my change https://www.ndviet.org/docker-selenium/ 
or download directly https://github.com/ndviet/docker-selenium/releases/download/selenium-grid-0.24.1/selenium-grid-0.24.1.tgz

**Comment by maksim-paskal** [comment]
@vietnd96 thanks, your changes also is working for me. Go ahead, it will be a fix to your previous contribution.

I don't use autoscaling, I don't tested, but pod, svc and ingress objects its fine to me.

**Comment by VietND96** [comment]
@maksim-paskal, thank you for your confirmation!

**Comment by VietND96** [comment]
Hi @maksim-paskal, the fix is released in chart version 0.25.1

**Comment by maksim-paskal** [comment]
@vietnd96 @diemol Thanks
